### PR TITLE
Adding Package `peewee`

### PR DESCRIPTION
**Pull request by michaelweinold**
### Description

I would like to add the [peewee](https://docs.peewee-orm.com/en/latest/) ORM. Since I have some trouble setting up the local build environment (cf. https://github.com/pyodide/pyodide/issues/3896), I hope this draft will provide a starting point for community members with more WASM experience.

With the help of the emscripten-forge team, I already added the package there: https://github.com/emscripten-forge/recipes/issues/295. [Compare also the recipe.yaml on emscripten-forge.](https://github.com/emscripten-forge/recipes/pull/322/files#diff-36637787b615de7ad0f9aa52669d570eb6c45d4cd357f2866ee60b45dcbc823f)

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests

### CODE REVIEW

**Comment by hoodmane** [comment]
Test failures here are flakes. It looks like it imports successfully. Could you add a simple test to make sure that it at least slightly functions?
For reference, [here's sqlalchemy's test](https://github.com/pyodide/pyodide/blob/main/packages/sqlalchemy/test_sqlalchemy.py). Once you do that we can merge it.

**Comment by michaelweinold** [comment]
Thank you for following up so quickly, @hoodmane! I added a simple test - could you have a look?

File: packages/peewee/test_peewee.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import os
```
Comment (hoodmane) [inline packages/peewee/test_peewee.py:1]: You'll need to:
1. Import `run_in_pyodide` at top level
2. Move other imports into the test function.

Globals in the file are not visible inside the function.

**Comment by michaelweinold** [comment]
I updated the test file accordingly and added a changelog entry.

**Comment by hoodmane** [comment]
Test is failing with:
```py
NameError: name 'database' is not defined
```
due to some weirdness about the way that Python class scopes work...

**Comment by michaelweinold** [comment]
That's odd - but as far as I understand it, that has nothing to do with the test or the recipe? Is this something that you would need to fix, or should I investigate further?

**Comment by hoodmane** [comment]
The actual traceback is:
```
  File "/lib/python311.zip/_pyodide/_base.py", line 468, in eval_code
    .run(globals, locals)
     ^^^^^^^^^^^^^^^^^^^^
  File "/lib/python311.zip/_pyodide/_base.py", line 310, in run
    coroutine = eval(self.code, globals, locals)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<exec>", line 15, in <module>
  File "/lib/python3.11/site-packages/peewee.py", line 3155, in __enter__
    self.connect()
  File "/lib/python3.11/site-packages/peewee.py", line 3187, in connect
    self._state.set_connection(self._connect())
                               ^^^^^^^^^^^^^^^
  File "/lib/python3.11/site-packages/peewee.py", line 3530, in _connect
    raise ImproperlyConfigured('SQLite driver not installed!')
peewee.ImproperlyConfigured: SQLite driver not installed!
```

**Comment by michaelweinold** [comment]
So the above test failure has been resolved - but the CI tests still give: `FAILED peewee/test_peewee.py::peewee[chrome] - ModuleNotFoundError: There was a problem with unpickling the return value/exception from your pyodide environment.`

**Comment by hoodmane** [comment]
I guess we should maybe fix pytest-pyodide and come back to this since it's not very ergonomic fixing errors like this.

**Comment by michaelweinold** [comment]
Sounds good! Should I open an issue? It seems that this is affecting other PRs also:

- https://github.com/pyodide/pyodide/pull/3897
- https://github.com/pyodide/pyodide/pull/3910

**Comment by michaelweinold** [comment]
@ryanking13 following [your suggestion in the test issue](https://github.com/pyodide/pytest-pyodide/issues/92#issuecomment-1582223731), the error [now reads](https://app.circleci.com/pipelines/github/pyodide/pyodide/5926/workflows/4566d5c0-72fe-42b5-8da5-002a5ad27def/jobs/75165):

```
---------------------------- Captured stdout setup -----------------------------
Spawning webserver at http://127.0.0.1:43205 (see logs in /tmp/tmpu981tdzo/http-server.log)
----------------------------- Captured stdout call -----------------------------
JavascriptException message:  Traceback (most recent call last):
  File "/lib/python311.zip/_pyodide/_base.py", line 468, in eval_code
    .run(globals, locals)
     ^^^^^^^^^^^^^^^^^^^^
  File "/lib/python311.zip/_pyodide/_base.py", line 310, in run
    coroutine = eval(self.code, globals, locals)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<exec>", line 4, in <module>
ModuleNotFoundError: The module 'peewee' is included in the Pyodide distribution, but it is not installed.
You can install it by calling:
  await micropip.install("peewee") in Python, or
  await pyodide.loadPackage("peewee") in JavaScript
See https://pyodide.org/en/stable/usage/loading-packages.html for more details.

--------------------------- Captured stdout teardown ---------------------------

------------ generated xml file: /root/repo/test-results/junit.xml -------------
============================= slowest 50 durations =============================
2.95s setup    packages/peewee/test_peewee.py::test_peewee[chrome]
0.24s teardown packages/peewee/test_peewee.py::test_peewee[chrome]
0.07s call     packages/peewee/test_peewee.py::test_peewee[chrome]
=========================== short test summary info ============================
FAILED peewee/test_peewee.py::peewee[chrome] - pytest_pyodide.runner.JavascriptException: PythonError: Traceback (most recent call last):
  File "/lib/python311.zip/_pyodide/_base.py", line 468, in eval_code
    .run(globals, locals)
     ^^^^^^^^^^^^^^^^^^^^
  File "/lib/python311.zip/_pyodide/_base.py", line 310, in run
    coroutine = eval(self.code, globals, locals)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<exec>", line 4, in <module>
ModuleNotFoundError: The module 'peewee' is included in the Pyodide distribution, but it is not installed.
You can install it by calling:
  await micropip.install("peewee") in Python, or
  await pyodide.loadPackage("peewee") in JavaScript
See https://pyodide.org/en/stable/usage/loading-packages.html for more details.

    at new_error (http://127.0.0.1:43205/pyodide.asm.js:9:12804)
    at http://127.0.0.1:43205/pyodide.asm.wasm:wasm-function[289]:0x153055
    at http://127.0.0.1:43205/pyodide.asm.wasm:wasm-function[290]:0x15315a
    at Module._pythonexc2js (http://127.0.0.1:43205/pyodide.asm.js:9:644050)
    at Module.callPyObjectKwargs (http://127.0.0.1:43205/pyodide.asm.js:9:78373)
    at Module.callPyObject (http://127.0.0.1:43205/pyodide.asm.js:9:78535)
    at p.apply (http://127.0.0.1:43205/pyodide.asm.js:9:93332)
    at Object.apply (http://127.0.0.1:43205/pyodide.asm.js:9:91574)
    at Object.runPython (http://127.0.0.1:43205/pyodide.asm.js:9:113202)
    at run (eval at executeAsyncScript (:443:26), <anonymous>:6:22)
    at eval (eval at executeAsyncScript (:443:26), <anonymous>:19:40)
    at eval (eval at executeAsyncScript (:443:26), <anonymous>:37:15)
    at eval (eval at executeAsyncScript (:443:26), <anonymous>:39:4)
    at executeAsyncScript (<anonymous>:443:47)
    at <anonymous>:458:29
    at callFunction (<anonymous>:338:22)
    at <anonymous>:351:23
    at <anonymous>:352:3
============================== 1 failed in 4.01s ===============================
```

Any suggestions on how to proceed?

**Comment by hoodmane** [comment]
You need `selenium.load_package("peewee")` first.

File: packages/peewee/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package:
+  name: peewee
+  version: 3.16.2
+  top-level:
+    - peewee
+requirements:
+  host:
+    - cffi
+  run:
+    - pysqlite3
```
Comment (hoodmane) [inline packages/peewee/meta.yaml:10]: Maybe:
```suggestion
    - sqlite3
```

**Comment by michaelweinold** [comment]
✅ Tests are now all passing. Many thanks for your support - it has been a very pleasant experience!

**Comment by michaelweinold** [comment]
✅ All checks are now passing. Many thanks for your help - it has been a most pleaseant experience.

**Comment by hoodmane** [comment]
Thanks @michaelweinold!

**Pull request merged**